beacon-chain/verification/error.go

```diff
@@ -36,3 +39,12 @@ func (ve VerificationMultiError) Failures() map[Requirement]error {
 func newVerificationMultiError(r *results, err error) VerificationMultiError {
 	return VerificationMultiError{r: r, err: err}
 }
+
+// VerifiedROBlobError can be used by methods that have a VerifiedROBlob return type but do not have permission to
+// create a value of that type in order to generate an error return value.
+func VerifiedROBlobError(err error) (blocks.VerifiedROBlob, error) {
+	if err == nil {
+		panic("VerifiedROBlobError used to create a VerifiedROBlob without a checkable error.")
```

Comment:
:scream: do we have to panic?? :scream:

Reply:
My reasoning was that we should only call this function when there's an error, because we want to make sure that this isn't accidentally used to get an empty VerifiedROBlob that gets used somehow, and that a panic would make a test fail... But you're right to flag this, because I suppose some code could be written that uses this incorrectly, and somehow doesn't fall under test coverage, which would be quite dangerous! A better alternative is to turn this into a custom error:

```
var errVerifiedROBlobErrorNil = errors.New(VerifiedROBlobError used to create a VerifiedROBlob without a checkable error.")

func VerifiedROBlobError(err error) (blocks.VerifiedROBlob, error) {
	if err == nil {
	    return ..., errVerifiedROBlobErrorNil
	    ...
```